### PR TITLE
Covid19 Screener

### PR DESCRIPTION
**Pull request by mchelen-gov**
## Description

covid19 screener for VAMCs (uses react hooks to store state)

## Screenshots
- [mobile](https://user-images.githubusercontent.com/63597655/82254768-7be81800-9921-11ea-8f39-ef8a3d67e71e.png)

## Acceptance criteria
https://github.com/department-of-veterans-affairs/va.gov-team/issues/9207

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/coronovirus-screener/tests/00.coronovirus-screener.e2e.spec.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const Timeouts = require('../../../platform/testing/e2e/timeouts');
+const E2eHelpers = require('../../../platform/testing/e2e/helpers');
+const manifest = require('../manifest.json');
+
+module.exports = E2eHelpers.createE2eTest(client => {
+  client
+    .url(`${E2eHelpers.baseUrl}/coronovirus-screener`)
```
Comment (batemapf) [inline src/applications/coronovirus-screener/tests/00.coronovirus-screener.e2e.spec.js:7]: should this be `/covid19screener`?

File: src/applications/coronovirus-screener/components/FormResult.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        </p>
+        <p>Thank you for helping us protect you and others during this time.</p>
+      </div>
+    </div>
+  );
+
+  const fail = (
+    <div>
+      <h2 className="vads-u-font-size--h1">More screening needed</h2>
+      <h3>Valid for: {moment().format('MMMM Do YYYY, h:mm:ss a')}</h3>
```
Comment (batemapf) [inline src/applications/coronovirus-screener/components/FormResult.jsx:27]: Mind dropping the seconds precision here?

**Comment by batemapf** [comment]
I'm thinking the path for the app should change from `src/applications/coronavirus-screener` to `src/applications/covid19screener` for consistency sake.

File: src/applications/coronovirus-screener/components/FormResult.jsx (hunk 2 above)
Comment (mchelen-gov) [reply src/applications/coronovirus-screener/components/FormResult.jsx:27]: yes... for all our E2E tests 😁

File: src/applications/coronovirus-screener/tests/00.coronovirus-screener.e2e.spec.js (hunk 1 above)
Comment (mchelen-gov) [reply src/applications/coronovirus-screener/tests/00.coronovirus-screener.e2e.spec.js]: yes... for all our E2E tests 😁

File: src/applications/coronovirus-screener/components/FormResult.jsx (hunk 2 above)
Comment (mchelen-gov) [reply src/applications/coronovirus-screener/components/FormResult.jsx:27]: done!

**Review by stephenBarrs — APPROVED** [review]
Looks good for MVP

**Comment by mchelen-gov** [comment]
> I'm thinking the path for the app should change from `src/applications/coronavirus-screener` to `src/applications/covid19screener` for consistency sake.

yeah you might be right, i was trying to stay consistent with chatbot and consider the URL secondary: https://github.com/department-of-veterans-affairs/vets-website/tree/master/src/applications/coronavirus-chatbot

but maybe it introduces more confusion than its worth?

**Comment by batemapf** [comment]
ahh that's a good point. we are supposed to be using coronavirus to describe the current plight of the world. only using covid in this case b/c it is shorter. so yeah, let's keep it coronavirus. good call.

**Review by batemapf — APPROVED** [review]

**Review by jhonny218 — APPROVED** [review]
LGTM

**Comment by rianfowler** [comment]
Feel free to set your team up as the [codeowner](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/codeowners.md) of this application.

**Comment by batemapf** [comment]
😻love this new process! thanks @rianfowler!

File: src/applications/coronovirus-screener/manifest.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "appName": "COVID-19 Screener",
+  "entryFile": "./app-entry.jsx",
+  "entryName": "covid19screener",
+  "rootUrl": "/covid19screener",
+  "production": false
```
Comment (bkjohnson) [inline src/applications/coronovirus-screener/manifest.json:6]: You may want to update `src/applications/registry.json` with `vagovprod` set to false for this application.

**Review by batemapf — CHANGES_REQUESTED** [review]
Path needs to be updated to `coronavirus-screener` from `coronovirus-screener`.

**Review by batemapf — APPROVED** [review]

File: src/applications/coronovirus-screener/manifest.json (hunk 3 above)
Comment (batemapf) [reply src/applications/coronovirus-screener/manifest.json]: thanks! any idea who wins if there is a conflict b/w registry.json and this manifest file?

Comment (mchelen-gov) [reply src/applications/coronovirus-screener/manifest.json]: does our app need an entry in `src/applications/registry.json`? it's not mentioned in [yeoman docs](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/platform/tools/generator/) i see it in [manual setup](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/common-tasks/new-application) though

Comment (bkjohnson) [reply src/applications/coronovirus-screener/manifest.json]: Right now the `registry.json` file is just used to [create the react landing pages](https://github.com/department-of-veterans-affairs/vets-website/blob/eccb647865b3d827f128f1392093b7a28771768f/src/site/stages/build/plugins/create-react-pages.js#L3-L24) for each application. One isn't overriding the other, so there shouldn't be any conflicts.

File: src/applications/coronavirus-screener/manifest.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "appName": "COVID-19 Screener",
+  "entryFile": "./app-entry.jsx",
+  "entryName": "covid19screener",
+  "rootUrl": "/covid19screener",
+  "production": false
```
Comment (bkjohnson) [inline src/applications/coronavirus-screener/manifest.json:6]: This won't prevent the app from going to prod.

**Review by bkjohnson — COMMENTED** [review]
I just want to point something out in case it has been overlooked.  Based on the `manifest.json`, it looks like you don't want this app to go to production yet.  However, the manifest _won't_ prevent that from happening.

From [the documentation](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/common-tasks/new-application):

image

File: src/applications/coronavirus-screener/manifest.json (hunk 4 above)
Comment (batemapf) [reply src/applications/coronavirus-screener/manifest.json]: thanks @bkjohnson! got this updated. our intention is that it _does_ go to prod. removed this attribute and [added the app into the registry](https://github.com/department-of-veterans-affairs/vets-website/pull/12673/commits/bbcef0937e2bb96d8aab96c5165a89319210fefa#diff-b2162e03ddfa38375fef380c0fe8cee5R9), w/ `vagovprod` to `true`.

**Review by batemapf — APPROVED** [review]

**Pull request merged**